### PR TITLE
Event Card Component - Start

### PR DESCRIPTION
**Pull request by k-kenney**
Hi everyone, I started the event card component today. After not coding properly for a while, I feel very rusty. 

Could you help me how with a few things?

1. Do you think there is a better way to organize my code? This was the best I could come up with for now. 
2. I'm not sure how to make the calendar component and event cards component line up next to one another.
3. I'm not sure if I was supposed to hardcode everything. That's what I've done until I know where/how to link things a bit better. I suspect that will require some JS though.

Thanks for your feedback!

You can't tell from this screenshot, but there is a white background with some opacity and rounded borders like it has in Figma

Screen Shot 2023-05-11 at 7 07 40 PM

### CODE REVIEW

**Comment by wJoenn** [comment]
Can you share a screenshot of your current work please ? 🙂

**Comment by wJoenn** [comment]
To answer one of your question already, no you should not hardcode the events.
You can use the existing events in your local db, I've added some to the seed and there is a `@events` variable in the `pages#home` action so you can run `rails db:seed` if you haven't already then iterate through each Event inside `@events` in your html

File: Gemfile.lock
Comment (kokoras88) [inline Gemfile.lock:1]: Could someone explain me please what you did Kayla did here?

**Review by kokoras88 — APPROVED** [review]

File: Gemfile.lock
Comment (wJoenn) [reply Gemfile.lock:1]: Why did you approve if you don't understand what she did ?

This PR is not finished, she wanted to ask for advice on what to do next. Please remove your approval and wait for the PR to be tagged as "waiting for review" 😄

Comment (k-kenney) [reply Gemfile.lock:1]: I'm almost there, Louis helped me figure out how to utilize all of the attributes and methods needed so I didn't have to hardcode anything :) 

I think the last step is to get the calendar and event cards to be positioned next to each other. Does anyone know how to use flexbox to get 2 partials to sit side-by-side? 

Screen Shot 2023-05-16 at 7 06 00 AM

**Comment by k-kenney** [comment]
Most recent version. The only thing I haven't figured out is how to get the arrows to change colors depending on the day of the week like our Figma has. For now, It just has a different color on hover. 

Screen Shot 2023-05-16 at 6 59 46 PM

**Comment by wJoenn** [comment]
You have to give them a different css every 3 event cards.
You can do that with nth-child.

You also need to make sure the height of the event cards isn't bigger than the height of the calendar.
If Chloe has 25 events we don't want to display a list of 25 events right ?
This needs to be scrollable

**Comment by k-kenney** [comment]
Good tips! I'll work on that.

**Comment by haqcked** [comment]
nice one @k-kenney, the event card looks real nice now. Im excited to see the scrollable event cards feature suggested by Louis 😀

and maybe this is the wrong place to suggest this, but before i forget id like to mention to @wJoenn the #handleClick icon prev & next are a bit small, i know its a minor detail for now but when i ran dev, to check it as u suggested, it felt really small. I almost didnt notice it at first 😅 perhaps we give it more size or perhaps a ring with a background color around it to make it more visible 😊

**Comment by k-kenney** [comment]
OK, almost there. I've been trying to figure out the right element/class to call for nth-child and I've had no luck. nth-child will work when I use li:nth-child(3n+1), but it only changes the text elements. When I target just the .fa-solid, and specify 3n+1 it changes all of the icons.  
https://github.com/wJoenn/YogaUnalome/assets/105766386/e4e85bc3-cab0-46a3-ac18-4d2b9fb5ef0d

**Comment by k-kenney** [comment]
I can't see to get just the right arrow icons to change color using nth-child or nth-type-of. I've watched a few videos and none of their tips seemed to solve the problem. Any suggestions? It might be better to go ahead a merge this so someone else can work on the nth-child part.

Screen Shot 2023-05-18 at 9 18 59 AM

**Comment by wJoenn** [comment]
You're using nth-child on the icon itself.
That means that inside each card you're trying to get the first, second, third, etc icon.
Except inside each card there's only 1 icon.
What you need to do is use nth-child on each card instead then select the icon
Something like `.card:nth-child(3) i { color: blue; }`

**Comment by k-kenney** [comment]
The latest - it's changing all icons to beige. Do you think it has something to do with font awesome icons?
Screen Shot 2023-05-18 at 9 18 59 AM

**Comment by wJoenn** [comment]
It's because you're adding the nth-child to the div.event-cards-content except there is only one .event-cards-content inside their parent element, the `<li>`
When nth-child does is it affect every element inside the element's parent element.
In your case you need to apply nth-child to the li.

**Comment by wJoenn** [comment]
Also please remove or style the scrollbar
image

**Comment by k-kenney** [comment]
I felt like I had tried every combination out there. Your previous comment proved helpful, @wJoenn . I styled the scrollbar, but it doesn't have to remain that way. 

Screen Shot 2023-05-19 at 10 19 30 AM

**Comment by wJoenn** [comment]
@k-kenney Did you notice the events don't appear in the calendar anymore ?
I know it's working on master so something must be wrong with your branch, can you fix that please ? 🙂

**Comment by k-kenney** [comment]
What do we think now? 

Screen Shot 2023-05-19 at 4 27 33 PM

File: db/seeds.rb
Comment (haqcked) [inline db/seeds.rb:13]: is this right with 2 title: ?

File: app/views/pages/partials/_event-card.html.erb
Comment (haqcked) [inline app/views/pages/partials/_event-card.html.erb:13]: indentation: i tag with the closing tag above it should be in the same level

Comment (wJoenn) [reply app/views/pages/partials/_event-card.html.erb:13]: @haqcked when you spot mistakes like this one don't hesitate to mark your review as `Request changes`
image

Comment (wJoenn) [reply app/views/pages/partials/_event-card.html.erb:13]: You can also change the tag of the PR to be `requires changes` instead of `waiting for review`

**Comment by k-kenney** [comment]
Good catch, @haqcked. Thanks for your feedback. I've corrected them.

File: app/views/pages/partials/_calendar.html.erb
Comment (cipriancrisan) [inline app/views/pages/partials/_calendar.html.erb:1]: I see that here the only difference is an indentation of the code to the right.
Could you please just indent it back one space to the left? (lines 1 to 37)

File: app/frontend/stylesheets/pages/partials/_event-card.scss
Comment (cipriancrisan) [inline app/frontend/stylesheets/pages/partials/_event-card.scss:30]: I have just learned this info, the power of SCSS is that we can nest the different selectors under one selector.
For example, in your `_event.html.erb` you have a section with the id `#event-card-container`. It means that you can add the properties to your `#event-card-container` and underneath you can nest the following selectors: `.event-cards` and `.event-cards-content`. Also as you can nest under `.event-cards-content` the h3 element and :hover property.
```
.event-cards-content {

  &:hover {
  }

  h3 {
  }
}
```
you can look at the `_calendar.scss` partial

**Review by cipriancrisan — CHANGES_REQUESTED** [review]
Hello Kayla, please take a look at the comments I have written in the different files in this PR. If you have any questions please let me know.

**Comment by wJoenn** [comment]
@cipriancrisan Please make sure to publish every comments as one single review and select the code proprelly next time.


https://github.com/wJoenn/YogaUnalome/assets/75388869/48eeaf97-8f3c-45ff-9664-847f8bd48783

File: app/frontend/stylesheets/pages/partials/_event-card.scss
Comment (k-kenney) [reply app/frontend/stylesheets/pages/partials/_event-card.scss:31]: Really useful feedback, thank you! I didn't know that about scss, I appreciate you sharing.

**Review by haqcked — APPROVED** [review]
LGTM

File: app/frontend/stylesheets/pages/_home.scss
Comment (wJoenn) [inline app/frontend/stylesheets/pages/_home.scss:21]: Can you move this inside the #planning proprety please.
There's also a trailing empty line that you can remove at line 21

File: app/frontend/stylesheets/pages/_index.scss
Comment (wJoenn) [inline app/frontend/stylesheets/pages/_index.scss:3]: Please move the event-card import between the calendar and the membership and rename the file `_event_cards.scss`

File: app/models/event.rb
Comment (wJoenn) [inline app/models/event.rb:1]: Validate the presence of the title please

**Review by wJoenn — CHANGES_REQUESTED** [review]
I'd like you to tweak the design a bit by

- [x] changing the cursor on the arrow buttons
- [x] make the scrollbar invisible when I'm not hovering the event-cards
- [x] I'd also like the scrollbar to be a bit thinner.

image


Finally can you work on a javascript feature that would scroll the event-cards container to show the correct day if I'm clicking on the calendar.
Let me explain a bit :
When Chloe will be using her website she will have 30 to 40 events at all times.
Having to scroll them to look for the one you want is not a great UX.
Browsing through the calendar is easier and more intuitive.
In this screenshot for example
image

I would that the event-cards container to scroll down enough to show the Event on the 18th of june when I click on the 18th of june in the calendar

**Review by wJoenn — CHANGES_REQUESTED** [review]
I'd like you to tweak the design a bit by

- [ ] changing the cursor on the arrow buttons
- [ ] make the scrollbar invisible when I'm not hovering the event-cards
- [ ] I'd also like the scrollbar to be a bit thinner.

image


Finally can you work on a javascript feature that would scroll the event-cards container to show the correct day if I'm clicking on the calendar.
Let me explain a bit :
When Chloe will be using her website she will have 30 to 40 events at all times.
Having to scroll them to look for the one you want is not a great UX.
Browsing through the calendar is easier and more intuitive.
In this screenshot for example
image

I would that the event-cards container to scroll down enough to show the Event on the 18th of june when I click on the 18th of june in the calendar

**Review by wJoenn — APPROVED** [review]

**Pull request merged**
